src/libs/sso-utils.js

```diff
@@ -0,0 +1,272 @@
+//
+// Reggie
+//
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Created by Shelly Xue Han on 2019-01-08.
+//
+
+'use strict';
+
+import request from 'request-promise-native';
+import url from 'url';
+import { SSO_SUB_URI, SSO_REQUEST } from '../constants';
+
+export const checkCredentialValid = credentials => {
+  if (!credentials.uri) {
+    throw new Error('A URL must be provided');
+  }
+  if (!(credentials.token || (credentials.username && credentials.password))) {
+    throw new Error('An auth info must be provided, such as username-password pair or token.');
+  }
+};
+
+export const checkUserProfile = (userInfo, userGroups = [], userIdps = []) => {
+  let valid = false;
+  try {
```

Comment:
why try catch? doesn't seem like anything will throw here

Reply:
right, updating that